fedimint-api/src/macros.rs

```diff
@@ -100,6 +139,75 @@ macro_rules! _dyn_newtype_define_inner {
     };
 }
 
+#[macro_export]
+macro_rules! _dyn_newtype_define_with_instance_id_inner {
+    (   $(#[$outer:meta])*
+        $vis:vis $name:ident($container:ident<$trait:ident>)
+    ) => {
+        $(#[$outer])*
+        $vis struct $name($container<dyn $trait + Send + Sync + 'static>, ::fedimint_api::core::ModuleInstanceId);
+
```

Comment:
I'm happy you were able to fit the instance id in there, so modules don't need to worry about it.